File: z-clients/javascript/src/models/listResponseSvixPollerOut.ts
// this file is @generated
import {
    type SvixPollerOut,
    SvixPollerOutSerializer,
} from './svixPollerOut';

export interface ListResponseSvixPollerOut {
    data: SvixPollerOut[];
    iterator?: string | null;
    prevIterator?: string | null;
    done: boolean;
}

export const ListResponseSvixPollerOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): ListResponseSvixPollerOut {
        return {
            data: object['data'].map((item: SvixPollerOut) => SvixPollerOutSerializer._fromJsonObject(item)),
            iterator: object['iterator'],
            prevIterator: object['prev_iterator'],
            done: object['done'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: ListResponseSvixPollerOut): any {
        return {
            'data': self.data.map((item) => SvixPollerOutSerializer._toJsonObject(item)),
            'iterator': self.iterator,
            'prev_iterator': self.prevIterator,
            'done': self.done,
        };
    }
}

File: z-clients/rust/src/models/list_response_svix_poller_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::svix_poller_out::SvixPollerOut;

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct ListResponseSvixPollerOut {
    pub data: Vec<SvixPollerOut>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub iterator: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub prev_iterator: Option<String>,

    pub done: bool,
}

impl ListResponseSvixPollerOut {
    pub fn new(data: Vec<SvixPollerOut>, done: bool) -> Self {
        Self {
            data,
            iterator: None,
            prev_iterator: None,
            done,
        }
    }

    pub fn with_iterator(mut self, value: impl Into<Option<String>>) -> Self {
        self.iterator = value.into();
        self
    }

    pub fn with_prev_iterator(mut self, value: impl Into<Option<String>>) -> Self {
        self.prev_iterator = value.into();
        self
    }
}
